### PR TITLE
Refactor drakvuf config to a single class

### PR DESCRIPTION
**Pull request by msm-cert**
I'm proud of my new creation

### CODE REVIEW

**Comment by msm-cert** [comment]
Hypothetically we could change self.config type to drakconifg maybe? But I'm not sure if that's "clean", I'd like to avoid changing karton internals too much.

File: drakrun/drakrun/main.py [hunk 1]
```diff
@@ -127,6 +127,81 @@ def wrapper(self: Karton, *args, **kwargs):
...
 
+class DrakrunConfig(Config):
+    """A typed wrapper for a Drakrun config. In the future, this will be handled
+    by Karton's typedconfig more directly
+    """
+
+    def __init__(self, config: Config) -> None:
+        super().__init__(config)
+
+    def get(self, option, fallback) -> Any:
```
Comment (nazywam) [inline drakrun/drakrun/main.py:138]:
```suggestion
    def get(self, option: str, fallback: Any) -> Any:
```

File: drakrun/drakrun/main.py [hunk 2]
```diff
@@ -127,6 +127,81 @@ def wrapper(self: Karton, *args, **kwargs):
...
+        return self.config.config.get("drakrun", option, fallback=fallback)
+
+    def getint(self, option, fallback) -> int:
```
Comment (nazywam) [inline drakrun/drakrun/main.py:141]:
```suggestion
    def getint(self, option: str, fallback: int) -> int:
```

File: drakrun/drakrun/main.py [hunk 3]
```diff
@@ -127,6 +127,81 @@ def wrapper(self: Karton, *args, **kwargs):
...
+        return self.config.config.getint("drakrun", option, fallback=fallback)
+
+    def getboolean(self, option, fallback) -> bool:
```
Comment (nazywam) [inline drakrun/drakrun/main.py:144]:
```suggestion
    def getboolean(self, option: str, fallback: bool) -> bool:
```

File: drakrun/drakrun/main.py [hunk 4]
```diff
@@ -127,6 +127,81 @@ def wrapper(self: Karton, *args, **kwargs):
...
+        return self.config.config.getboolean("drakrun", option, fallback=fallback)
+
+    @property
+    def out_interface(self) -> str:
+        return self.get("out_interface", "")
```
Comment (nazywam) [inline drakrun/drakrun/main.py:149]: I think I'd annotate the fallback arguments here and in the rest of the properties for better clarity, so:

```python
return self.get("out_interface", fallback="")
```

**Review by nazywam — APPROVED** [review]

**Pull request merged**